src/cocotb/handle.py

```diff
@@ -476,8 +476,30 @@ def _child_path(self, name: str) -> str:
     def _sub_handle_key(self, name: str) -> str:
         return name
 
-    def _get_handle_by_key(self, key: str) -> Optional[simulator.gpi_sim_hdl]:
-        return self._handle.get_handle_by_name(key)
+    def _get_handle_by_key(
+        self, key: str, native=False
+    ) -> Optional[simulator.gpi_sim_hdl]:
```

Comment:
Something like this is fine. The additional argument would default to `AUTO` which would pass through to `gpi_get_handle_by_name`. The other discovery methods: `__getitem__`, `_id`, and `__getattr__` would just use such a function with the defaulted `AUTO`. However, I think this might be the wrong function to accomplish that, as it should hook into the sub object cache.

Maybe a new function like `_get` to parallel `dict.get`? Then refactor `__getitem__`, `_id`, and `__getattr__` to use that instead?

Reply:
I am very much in favor of using _get to mimic dict.get, this is exactly what I had in mind originally when I opened the issue.

the refactoring should be easy to do. 

Same question as in the other comment: how do we want to handle the possible values for the "discovery=" parameter. (is that what we want to call it?)

Reply:
went with implementing _get